### PR TITLE
Cv2 4606 delete message

### PR DESCRIPTION
**Pull request by DGaffney**
If we don't delete the existing message this code just has the net effect of *compounding the same task repeatedly* which is definitely not what we want! Increment the count, delete the old job, and add a new job.

### CODE REVIEW

File: lib/queue/worker.py [hunk 1]
```diff
@@ -144,4 +144,5 @@ def increment_message_error_counts(self, messages_with_queues: List[Tuple]):
             else:
                 updated_message = schemas.parse_message(message_body)
                 updated_message.retry_count = retry_count
+                queue.delete_messages(Entries=[self.delete_message_entry(message)])
```
Comment (skyemeedan) [inline lib/queue/worker.py:147]: this isn't including the idx (which is mapped to the id) is it needed?

**Review by skyemeedan — COMMENTED** [review]
lol, this makes sense. Does it work locally?

File: lib/queue/worker.py (hunk 1 above)
Comment (DGaffney) [reply lib/queue/worker.py:147]: yep - idx is only needed as some unique identifier in a list of items - and on the other side of this, it just defaults to 0 if not passed along, so this works fine!

**Comment by DGaffney** [comment]
> lol, this makes sense. Does it work locally?

lol, yep - and yes I did test this on the QA box directly to make sure that message deletion would work, as I was figuring out what the error was in this latest deploy

**Review by skyemeedan — APPROVED** [review]

**Pull request merged**